src/Authenticator/FormAuthenticator.php

```diff
@@ -100,14 +105,76 @@ protected function _checkLoginUrl(ServerRequestInterface $request)
         $loginUrl = $this->getConfig('loginUrl');
 
         if (!empty($loginUrl)) {
-            if (is_array($loginUrl)) {
-                $loginUrl = Router::url(['_base' => false] + $loginUrl);
-                $this->setConfig('loginUrl', $loginUrl);
+            if (is_string($loginUrl) && $this->_compareStringUrl($loginUrl, $request->getUri()->getPath())) {
+                return true;
             }
 
-            return strcasecmp($request->getUri()->getPath(), $loginUrl) === 0;
+            return $this->_compareArrayUrl($loginUrl, $request);
```

Comment:
Do we need array URL support? I'd like to believe that simple string URLs and regex patterns should cover the bulk of use cases.

Reply:
> Do we need array URL support?

I don't think we do. 

I can handle string URL approach if you @burzum and @markstory don't mind.